PR: Migrated to Vite, Revamped UI & Improved Security

HOTHEAD01TH opened the pull request:
What’s Changed?
- Migrated from CRA to Vite → Cold start `(25s → 2s`), hot reload `(<1s`), and build times `(60s → 15s)`.
- Revamped UI → More modern, scalable, and responsive.
- Removed deprecated packages → Better stability & security.
- Improved security → Moved hardcoded API keys to `.env`.
- Optimized assets → Replaced image logos with `React Icons` for better performance.
- Better error handling → Added `react-toastify` for cleaner alerts.
- Added regex validation → Improved link handling in description.
- The app is now faster, safer, and more scalable. 🚀
- live link : https://free-youtube-player.vercel.app/

HOTHEAD01TH committed:
feat: migrated CRA to Vite, moved API to env, and designed a cool landing page

HOTHEAD01TH committed:
revamped header/body/footer , better error handling, more playlists

HOTHEAD01TH committed:
Merge branch 'main' of https://github.com/HOTHEAD01TH/youtube-player

HOTHEAD01TH committed:
minor fix

HOTHEAD01TH committed:
Removed VITE_ prefix from environment variables to prevent client-side exposure

hotheadhacker commented:
@HOTHEAD01TH merged, added new url, also do we need to set up new deployment pipelines?

HOTHEAD01TH commented:
> @HOTHEAD01TH merged, added new url, also do we need to set up new deployment pipelines?

@hotheadhacker No additional changes to the pipeline are needed at the moment. However, since you're using Netlify, you might need to add the YOUTUBE_API_KEY in the Netlify environment variable dashboard and update the root directory from frontend to empty (since the main project is now the root dir). Since you've updated the live link, I've already added the API key in my Vercel deployment, so no further action is needed
